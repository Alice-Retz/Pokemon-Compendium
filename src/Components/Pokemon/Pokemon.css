img {
	max-width: 150px;
	height: auto;
}

.pokeEntry {
	display: flex;
	flex-direction: column;
	align-items: center;
	width: 300px;
	margin: 5px;
	background-color: aliceblue;
}
