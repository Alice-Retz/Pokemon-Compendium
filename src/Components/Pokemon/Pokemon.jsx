import PropTypes from 'prop-types';
import './Pokemon.css';

export default function Pokemon({ pokemon }) {
	const {
		name,
		pic,
		type1,
		type2,
		ability1,
		ability2,
		hiddenAbility,
		eggGroup1,
		eggGroup2,
	} = pokemon;

	const hasSecondType = type2 !== 'NA';
	const hasSecondAbility = ability2 !== 'NA';
	const hasSecondEggGroup = eggGroup2 !== 'NA';

	return (
		<article className='pokeEntry'>
			<h2>{name}</h2>
			<img src={pic} alt={`${name}`} />
			<div className='details'>
				<p>
					{type1}
					{hasSecondType && `/${type2}`}
				</p>
				<p>
					{ability1}
					{hasSecondAbility && `/${ability2}`}
					{hiddenAbility}
				</p>
				<p>
					{eggGroup1}
					{hasSecondEggGroup && `/${eggGroup2}`}
				</p>
			</div>
		</article>
	);
}

Pokemon.propTypes = {
	pokemon: PropTypes.shape({
		name: PropTypes.string.isRequired,
		pic: PropTypes.string.isRequired,
		type1: PropTypes.string.isRequired,
		type2: PropTypes.string.isRequired,
		ability1: PropTypes.string.isRequired,
		ability2: PropTypes.string.isRequired,
		hiddenAbility: PropTypes.string.isRequired,
		eggGroup1: PropTypes.string.isRequired,
		eggGroup2: PropTypes.string.isRequired,
	}).isRequired,
};
