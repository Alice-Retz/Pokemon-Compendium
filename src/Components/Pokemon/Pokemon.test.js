import { screen, render } from '@testing-library/react';
import Pokemon from './Pokemon';

it('Should render a pokemon', () => {
	render(
		<Pokemon
			pokeEntry={{
				name: 'butterfree',
				pic: 'http://assets.pokemon.com/assets/cms2/img/pokedex/full/012.png',
				type1: 'bug',
				type2: 'flying',
				ability1: 'compund-eyes',
				ability2: 'NA',
				hiddenAbility: 'tinted-lens',
				eggGroup1: 'bug',
				eggGroup2: 'bug',
			}}
		/>
	);

	const pokemon = screen.getByText(/butterfree/);
	expect(pokemon).toMatchSnapshot();
});
