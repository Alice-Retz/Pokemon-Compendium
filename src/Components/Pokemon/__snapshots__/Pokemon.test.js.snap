// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a pokemon 1`] = `
<h2>
  butterfree
</h2>
`;
